builtin/providers/clc/resource_clc_load_balancer_pool.go

```diff
@@ -0,0 +1,189 @@
+package clc
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	clc "github.com/CenturyLinkCloud/clc-sdk"
+	"github.com/CenturyLinkCloud/clc-sdk/lb"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceCLCLoadBalancerPool() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCLCLoadBalancerPoolCreate,
+		Read:   resourceCLCLoadBalancerPoolRead,
+		Update: resourceCLCLoadBalancerPoolUpdate,
+		Delete: resourceCLCLoadBalancerPoolDelete,
+		Schema: map[string]*schema.Schema{
+			// pool args
+			"port": &schema.Schema{
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"data_center": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"load_balancer": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"method": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "roundRobin",
+			},
+			"persistence": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "standard",
+			},
+			"nodes": &schema.Schema{
+				Type:     schema.TypeList,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeMap},
+			},
+		},
+	}
+}
+
+func resourceCLCLoadBalancerPoolCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clc.Client)
+	dc := d.Get("data_center").(string)
+	lbid := d.Get("load_balancer").(string)
+
+	s1 := d.Get("method").(string)
+	m := lb.LeastConn
+	if s1 == fmt.Sprintf("%v", lb.RoundRobin) {
+		m = lb.RoundRobin
+	}
+	s2 := d.Get("persistence").(string)
+	p := lb.Standard
+	if s2 == fmt.Sprintf("%v", lb.Sticky) {
+		p = lb.Sticky
+	}
+	r2 := lb.Pool{
+		Port:        d.Get("port").(int),
+		Method:      m,
+		Persistence: p,
+	}
+	lbp, err := client.LB.CreatePool(dc, lbid, r2)
+	if err != nil {
+		return fmt.Errorf("Failed creating pool under %v/%v: %v", dc, lbid, err)
+	}
+	d.SetId(lbp.ID)
+
+	nodes, err := parseNodes(d)
+	if err != nil {
+		return err
+	}
+
+	// HACK: create does not allow pegging nodes so do an immediate update
+	err = client.LB.UpdateNodes(dc, lbid, lbp.ID, nodes...)
+	if err != nil {
+		return fmt.Errorf("Failed setting nodes on %v: %v", lbp.ID, err)
+	}
+
+	return resourceCLCLoadBalancerPoolRead(d, meta)
+}
+
+func resourceCLCLoadBalancerPoolRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clc.Client)
+	dc := d.Get("data_center").(string)
+	lbid := d.Get("load_balancer").(string)
+	id := d.Id()
+	pool, err := client.LB.GetPool(dc, lbid, id)
+	if err != nil {
+		return fmt.Errorf("Failed fetching pool under %v/%v: %v", dc, lbid, err)
+	}
```

Comment:
Need to either recognize a NotFound / 404 error here and `d.SetId(""); return nil` or implement Exists() on the resource so you can assume the resource exists by the time you get to `Read()`.

Otherwise when a resource is deleted outside of Terraform we can't make progress because we keep getting the 404 error.